### DISCUSSION TITLE
Using ISIF=2 for Surface Calculations

### DISCUSSION THREAD

**Discussion by abhardwaj73** [materialsproject/pymatgen#4098] [Q&A]
Hello everyone,

I'm doing surface slab calculations for Calcite, Dolomite, and other related materials. I found out that various input set generators for these calculations like [`MVLSlabSet`](https://github.com/materialsproject/pymatgen/blob/f10ace8809765ce573fb334269ee199d59b2dbb8/pymatgen/io/vasp/sets.py#L2135)
uses ISIF=2 for relaxation. Is this the correct choice? I'm not sure since the stresses are not relaxed using ISIF=2.

One reason I could think of is the underlying bulk material will not allow the surface to relax fully (or at least the stresses).

Please let me know if using ISIF=2 for surface slab calculations is the correct choice.

**Comment by DanielYang59** [comment]
>  uses ISIF=2 for relaxation. Is this the correct choice? I'm not sure since the stresses are not relaxed using ISIF=2.

I'm afraid you might have misread the VASP manual because ["ISIF=2" does calculate stress tensors](https://www.vasp.at/wiki/index.php/ISIF).
> For ISIF ≥ 2, the stress tensor is calculated.

I'm by no means an expert, but what I usually do is using "ISIF=3" to relax the cell first (for slab with vacuum level), and then using "ISIF=2" to relax ionic positions. For a slab though, I usually [fix the cell along the vacuum axis during cell relaxation](https://github.com/Chengcheng-Xiao/VASP_OPT_AXIS). Alternatively you might try the [Birch–Murnaghan equation of state](https://en.wikipedia.org/wiki/Birch%E2%80%93Murnaghan_equation_of_state).
